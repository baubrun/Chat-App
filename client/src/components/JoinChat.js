import React, { useState } from "react";
import { Link } from "react-router-dom";

const JoinChat = () => {
  const [name, setName] = useState("");
  const [chatRoom, setChatRoom] = useState("");

  return (
    <div className="join-chat-container">
      <h1 className="app-title">ON PARLE</h1>
      <form className="join-chat">
        <div>
          <input
            className="join-chat"
            id="name-input"
            onChange={(event) => setName(event.target.value)}
            placeholder="Name"
            type="text"
            value={name}
          />
        </div>
        <div>
          <input
            className="join-chat mt-20"
            id="chatRoom-input"
            onChange={(event) => setChatRoom(event.target.value)}
            placeholder="Chat Room"
            type="text"
            value={chatRoom}
          />
        </div>
        <Link
          onClick={(event) =>
            !chatRoom || !name ? event.preventDefault() : ""
          }
          to={`/chat?name=${name}&chatRoom=${chatRoom}`}
        >
          <button className="join-chat button mt-20" type="submit">
            Join Chat
          </button>
        </Link>
      </form>
    </div>
  );
};

export default JoinChat;
