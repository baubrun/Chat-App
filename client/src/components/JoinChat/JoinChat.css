.button {
    border: none;
    background: #E26D5C;
    color: whitesmoke !important;
    border-radius: 5px;
    font-size: 1.2em;
    font-weight: bold;
    padding: 20px;
    text-transform: uppercase;
    text-decoration: none;
    width: 100%;
}


.join-chat {
    border-radius: 0;
    padding: 15px 20px;
    text-align: center;
    width: 60%;
}


.join-chat-container
{
    align-items: center;
    display: flex;
    flex-direction: column;
    height: 100vh;
    justify-content: center;
}


.join-chat::placeholder {
    font-size: 1.2rem;
}

.join-chat input {
    border-color: #E26D5C;
    font-size: 1.2em;
}

.mt-20 {
    margin-top: 20px;
}


