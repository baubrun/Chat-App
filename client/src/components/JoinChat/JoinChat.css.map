{
    "version": 3,
    "mappings": "ACAA,OAAO,CAAC,kEAAI;AACZ,OAAO,CAAC,6EAAI;ADUZ,AAAA,UAAU,CAAC;EACP,gBAAgB,ECLF,OAAO;EDMrB,MAAM,EAAE,GAAG,CAAC,KAAK,CCNH,OAAO;EDOrB,OAAO,EAAE,IAAI;EACb,MAAM,EAAE,GAAG;EACX,eAAe,EAAE,aAAa;EAC9B,cAAc,EAAE,MAAM;EACtB,KAAK,EAAE,GAAG;CACb;;AAKD,AAAA,eAAe,CAAC;EACZ,OAAO,EAAE,IAAI;EACb,aAAa,EAAE,CAAC;EAChB,cAAc,EAAE,MAAM;EACtB,UAAU,EAAE,MAAM;EAClB,KAAK,EAAE,GAAG;CA0Bb;;AA/BD,AAOI,eAPW,CAOX,KAAK,CAAC;EACF,YAAY,ECzBF,OAAO;ED0BjB,SAAS,EAAE,IAAI;EACf,MAAM,EAAE,MAAM;EACd,OAAO,EAAE,IAAI;CACZ;;AAZT,AAcI,eAdW,CAcX,MAAM,CAAC;EACH,MAAM,EAAE,IAAI;EACZ,UAAU,ECjCA,OAAO;EDkCjB,KAAK,EChCC,UAAU;EDiChB,aAAa,EAAE,GAAG;EAClB,SAAS,EAAE,KAAK;EAChB,WAAW,EAAE,IAAI;EACjB,UAAU,EAAE,IAAI;EAChB,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,SAAS;CAE5B;;AAzBL,AA2BI,eA3BW,CA2BX,aAAa,CAAA;EACT,SAAS,EAAE,KAAK;EAChB,UAAU,EAAE,MAAM;CACrB;;AAIL,AAAA,oBAAoB,CAAC;EACjB,WAAW,EAAE,MAAM;EACnB,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,MAAM,EAAE,KAAK;EACb,eAAe,EAAE,MAAM;CAC1B;;AAGD,AAAA,UAAU,AAAA,aAAa,CAAC;EACpB,SAAS,EAAE,MAAM;CACpB;;AAED,AAAA,gBAAgB,CAAC;EACb,KAAK,EC/DK,UAAU;CDgEvB;;AAED,AAAA,MAAM,CAAC;EACH,UAAU,EAAE,IAAI;CACnB",
    "sources": [
        "JoinChat.scss",
        "../../scss/variables.scss"
    ],
    "names": [],
    "file": "JoinChat.css"
}