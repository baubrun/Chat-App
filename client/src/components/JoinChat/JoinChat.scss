@import "../../scss/variables.scss";


// .chat-container {
//     align-items: center;
//     display: flex;
//     flex-direction: column;
//     height: 100vh;
//     justify-content: center;
// }

.container {
    background-color: $color-dark-pink;
    border: 2px solid $color-dark-pink;
    display: flex;
    height: 70%;
    justify-content: space-between;
    flex-direction: column;
    width: 70%;
}




.join-chat-form {
    display: flex;
    border-radius: 0;
    flex-direction: column;
    text-align: center;
    width: 60%;

    input {
        border-color: $color-dark-pink;
        font-size: 18px;
        margin: 10px 0;
        padding: 10px;
        }

    button {
        border: none;
        background: $color-dark-pink;
        color: $color-white;
        border-radius: 5px;
        font-size: 1.2em;
        font-weight: bold;
        margin-top: 10px;
        padding: 20px;
        text-transform: uppercase;
        // text-decoration: none;
    }

    ::placeholder{
        font-size: 1.2em;
        text-align: center;
    };
}


.join-chat-container {
    align-items: center;
    display: flex;
    flex-direction: column;
    height: 100vh;
    justify-content: center;
}


.join-chat::placeholder {
    font-size: 1.2rem;
}

#join-chat-title {
    color: $color-white;
}

.mt-20 {
    margin-top: 20px;
}