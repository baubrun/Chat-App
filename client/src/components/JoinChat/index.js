export {default as JoinChat} from "./JoinChat"