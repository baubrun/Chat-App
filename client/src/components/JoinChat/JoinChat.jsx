import React, { useState } from "react";
import { Title } from "../Title";
import { Button } from "../Button";
import "./JoinChat.css";
import { FormInput } from "../FormInput";
import { useHistory } from "react-router-dom";

const JoinChat = () => {
  const [name, setName] = useState("");
  const [chatRoom, setChatRoom] = useState("");
  let history = useHistory();

  const handleSubmit = (event) => {
    if (!chatRoom || !name) {
      return event.preventDefault();
    } else {
      history.push(`/chat?name=${name}&chatRoom=${chatRoom}`);
    }
  };

  return (
    <div className="join-chat-container">
      <Title id="join-chat-title" text="ON PARLE" />
      <form className="join-chat-form" onSubmit={handleSubmit}>
        <FormInput
          id="form-input"
          name="name"
          onChange={(event) => setName(event.target.value)}
          placeholder="Username"
          type="text"
          value={name}
        />
        <FormInput
          id="form-input"
          name="chatRoom"
          onChange={(event) => setChatRoom(event.target.value)}
          placeholder="Chat Room"
          type="text"
          value={chatRoom}
        />
        <Button type="submit" value="Join Chat" />
      </form>
    </div>
  );
};

export default JoinChat;
