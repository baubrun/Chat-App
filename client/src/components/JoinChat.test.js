import React from "react";
import { mount, shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import JoinChat from "./JoinChat";



const simulateChangeOnInput = (wrapper, inputSelector, givenValue) => {
  let input = wrapper.find(inputSelector);
  input.simulate("change", {
    target: { value: givenValue },
  });
  return wrapper.find(inputSelector);
};


describe("Join Chat", () => {

  it("inputs should get values submitted", () => {
    const wrapper = shallow(<JoinChat />);
    const nameInput = simulateChangeOnInput(wrapper, "input#name-input", "a");
    const chatRoomInput = simulateChangeOnInput(
      wrapper,
      "input#chatRoom-input",
      "asdf"
    );
    expect(nameInput.props().value).toEqual("a");
    expect(chatRoomInput.props().value).toEqual("asdf");
  });

});


describe('JoinChat Component', () => {
    const wrapper = shallow(<JoinChat />)
    expect(toJSON(wrapper)).toMatchSnapshot();
});