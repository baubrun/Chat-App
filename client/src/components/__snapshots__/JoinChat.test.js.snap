// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="join-chat-container"
>
  <h1
    className="app-title"
  >
    ON PARLE
  </h1>
  <form
    className="join-chat"
  >
    <div>
      <input
        className="join-chat"
        id="name-input"
        onChange={[Function]}
        placeholder="Name"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        className="join-chat mt-20"
        id="chatRoom-input"
        onChange={[Function]}
        placeholder="Chat Room"
        type="text"
        value=""
      />
    </div>
    <Link
      onClick={[Function]}
      to="/chat?name=&chatRoom="
    >
      <button
        className="join-chat button mt-20"
        type="submit"
      >
        Join Chat
      </button>
    </Link>
  </form>
</div>
`;
